system/jlib/jsecrets.cpp

```diff
@@ -1119,19 +1119,23 @@ static IPropertyTree * resolveLocalSecret(const char *category, const char * nam
     {
         if (entries->isDir())
             continue;
-        StringBuffer name;
-        entries->getName(name);
-        if (!validateXMLTag(name))
+        StringBuffer secretKeyName;   // filename is used as the secret key name
+        entries->getName(secretKeyName);
+        if (!validateXMLTag(secretKeyName))
             continue;
         MemoryBuffer content;
         Owned<IFileIO> io = entries->query().open(IFOread);
-        read(io, 0, (size32_t)-1, content);
+        read(io, 0, (size32_t)-1, content);   // contents of the file is the secret value
         if (!content.length())
             continue;
 
+        // Remove newline if present
+        if (content.toByteArray()[content.length()-1] == '\n')
+            content.setLength(content.length()-1);
```

Comment:
so dealing with trailing whitespace.. I think it should be more generalized.
could use StringBuffer::clip()